R/plot.R

```diff
@@ -68,57 +68,196 @@ fldts2df <- function(fldts, griddata)
-#' #' \code{\link{read.temperatures}}.
-#' #' @param nb Number of breaks in the color scale.  If nb < 2, use a smooth
-#' #' gradient.
-#' #' @param minval Lower limit of the color scale.  The default value was chosen
-#' #' to work well for fields of residuals from the mean temperature response.
-#' #' @param maxval Upper limit of the color scale.  The default value was chosen
-#' #' to work well for fields of residuals from the mean temperature response.
-#' #' @param legendstr String to use for the title of the graph legend
-#' #' @param palettestr String with the name of the ggplot2 color palette to use.
-#' #' Defaults to 'RdYlBu'. Options are Diverging: 'BrBG', 'PiYG', 'PRGn', 'PuOr',
-#' #' 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn', 'Spectral'; Qualitative: 'Accent',
-#' #' 'Dark2', 'Paired', 'Pastel1', 'Pastel2', 'Set1', 'Set2', 'Set3';
-#' #' Sequential: 'Blues', 'BuGn', 'BuPu', 'GnBu', 'Greens', 'Greys', 'Oranges',
-#' #' 'OrRd', 'PuBu', 'PuBuGn', 'PuRd', 'Purples', 'RdPu', 'Reds', 'YlGn',
-#' #' 'YlGnBu', 'YlOrBr', 'YlOrRd'.
-#' #' @param palettedir Numerical value, only accepts 1 or -1, determining the
-#' #' order of the color palette used. Defaults to -1.
-#' #' @export
-#' plot_field <- function(fld, griddata, nb=6, minval=-3.5, maxval=3.5, legendstr="Temperature (K)", palettestr = 'RdYlBu', palettedir = -1 )
-#' {
-#'     if(requireNamespace('gcammaptools')) {
-#'         tdf <- fld2df(fld, griddata)
-#'         ## TODO: make these options a little more customizable
-#'         if(nb < 2) {
-#'             gcammaptools::plot_GCAM_grid(tdf, col='value', extent=gcammaptools::EXTENT_WORLD,
-#'                                          legend=TRUE) +
-#'               ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir,
-#'                                             limits=c(minval, maxval), oob=scales::squish,
-#'                                             guide=ggplot2::guide_colorbar(title=legendstr, title.position='top'))
-#'         }
-#'         else {
-#'             ## Discretize the output values.
-#'             tdf$value <- minval +
-#'               findInterval(tdf$value, seq(minval, maxval, length.out=nb))/nb * (maxval-minval)
-#'             gcammaptools::plot_GCAM_grid(tdf, col='value', extent=gcammaptools::EXTENT_WORLD,
-#'                                          legend=TRUE) +
-#'               ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir, limits=c(minval,maxval),
-#'                                             guide=ggplot2::guide_colorbar(title=legendstr, title.position='top'))
-#'         }
-#'     }
-#'     else {
-#'         NULL
-#'     }
-#' }
+#' Plot a single field in matrix form
+#'
+#' Transform the field into a data frame using \code{\link{fld2df}} and plot
+#' using the \code{gcammaptools} package.  If \code{gcammaptools} isn't
+#' available, return \code{NULL}.  (TODO: make an ersatz plot if gcammaptools
+#' isn't there)
+#'
```

Comment:
These comments mention gcammaptools which is misleading because we are removing that dependency.